tools/bag_processing/scripts/merge_bags.py

```diff
@@ -77,12 +81,23 @@ def merge_bag(input_bag_prefix, merged_bag, only_loc_topics=False):
 
 
 if __name__ == "__main__":
-    parser = argparse.ArgumentParser()
-    parser.add_argument("input_bag_prefix")
-    parser.add_argument("--merged-bag", default="")
+    parser = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+    parser.add_argument(
```

Comment:
For improved flexibility, it seems like you could avoid requiring the bags to have a common prefix if you instead used `argparse` to accept a list of positional arguments using `nargs='+'` [1]. Then the user could easily recover the prefix behavior by specifying `<prefix>*`, which would be expanded by their shell.

(Sorry if this is a dupe. I tried to add a comment like this before, but it looked like it disappeared somehow.)

[1] https://docs.python.org/2.7/library/argparse.html#nargs

Reply:
I think in practice for us the bags all come with in the same directory, so this prevents having to type each of the prefixes by hand.  (The same comment applies to the convert all bayer images script).  I think for general purpose usage I could add the option to either manually list the prefixes/bags or allow them to be found assuming they are in the same directory, but I might leave that for future work if that's ok as I haven't had that use case yet.

Reply:
Done, added automatically finding prefixes as default behavior if no prefixes are provided